packages/data-broker/src/services/weather/WeatherService.ts

```diff
@@ -71,6 +90,8 @@ export class WeatherService extends Service {
 
       return this.getHistoricWeather(entities, startDate, endDate, apiResultTranslator);
     }
+
+    return resultFormat === 'analytics' ? EMPTY_ANALYTICS_RESULTS : ([] as EventResults);
```

Comment:
:exclamation: Impementation change

This is a breaking change with the purpose of using consistent types. The existing behaviour is returning `undefined` here, which will result in `[undefined]` being returned as the resulting analytics/events, which seems wrong.

I chose to return an empty response because this is what the other services do when no data source is found.